### DISCUSSION TITLE
Slots are not working in compoents, please help

### DISCUSSION THREAD

**Discussion by sreyansjain** [mhanberg/temple#220] [Q&A] [answered]
Hello,
I am trying to create a component using Temple.
Following is the code
```
defmodule DoorsWeb.CustomComponents do
  import Temple

  def my_temple_header(assigns) do
    temple do
      div do
        slot @inner_block
      end
    end
  end
end

```
I get an error like this - 
```
Compiling 107 files (.ex)
    error: undefined function render_slot/2 (expected DoorsWeb.CustomComponents to define such a function or for it to be imported, but none are available)
    │
  5 │     temple do
    │     ^^^^^^^^^
    │
    └─ lib/doors_web/components/custom_components.ex:5: DoorsWeb.CustomComponents.my_temple_header/1

```
I am using temple ver 0.12.0
elixir: 1.16.1 with OTP 26
Phoenix: 1.17.10

Please adivse if I am doing anything wrong. Thanks.

**Comment by mhanberg** [comment, marked as answer]
https://hexdocs.pm/temple/Temple.Component.html

You need to create a "component implementation"

**Reply by mhanberg** [reply]
Well if you are using LiveView, I think you just need to import Phoenix.Component

**Reply by sreyansjain** [reply]
I am not using LiveView I am using dead views or simple html views new heex things using 
```
use DoorsWeb, :html
```

When I import Phoenix.Component i still get an error message - 
```
undefined variable "changed" (context Phoenix.LiveView.Engine)
    (elixir 1.14.3) expanding macro: Kernel.var!/2
...
```

**Reply by sreyansjain** [reply]
I created my own implementation of Template.Component and components are working fine now. Thanks.
Awesome Library!
